### DISCUSSION TITLE
MAX3 - skipped packets

### DISCUSSION THREAD

**Discussion by githubDante** [johanmeijer/grott#210] [General]
Hi,

These look like valid records but are skipped due to the 03 value in the header

         - no matching record layout found, try generic
         - Record layout used :  T060103X
         - Growatt data decrypted V2
         - Grott Growatt data decrypted
         - Growatt plain data:
                 000d000602410103
 They are sent by the datalogger on reconnect (after restart of Grott in proxy mode). The header of the processed ones is
 
         - Record layout used :  T060104X
         - Growatt data decrypted V2
         - Grott Growatt data decrypted
         - Growatt plain data:
                 0085000602410104
 The check here:
 https://github.com/johanmeijer/grott/blob/83530c514f7aeeee7e3868e363b46e95b4f92e29/grottdata.py#L91
 marks them as invalid. The length of both is 585 bytes.  
Will have more info on the data inside the packet when finish the layout file, but at the moment it looks like a case which is not handled.

**Comment by johanmeijer** [comment]
I know.  The 03 records (and 29 for some inverters) are some kind of announcement  records with config info in It. The are only sent after a (re)connect.

I left them in on purposes while I still have decoding this records on my wishlist (gives eg hopefully info about inverter type). I think it does not hurts to have one of these records a day in the log and real decoding has not be done while no layout definition is available.

**Comment by githubDante** [comment]
Hi,

These indeed are announcement/report packets. Already working on a layout for the T060103X,
it should cover the SPH/SPF/TL3 inverters.

The packet contains information about the device:
 - date & time 
 - firmware version
 - device type
 - device model
 - different configuration settings

Also need help with this '&*6'. Don't know how must be presented:

43. | DTC | Device Type Code | &*6

as integer on TL3 the register shows value of 5000.

Would you like to add annotations in the json file which registers are writable ?!?
They can be helpful for the grott server.
